main/src/cgeo/geocaching/apps/navi/OsmAndApp.java

```diff
@@ -0,0 +1,97 @@
+import cgeo.geocaching.models.Waypoint;
+import cgeo.geocaching.utils.ProcessUtils;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class OsmAndApp extends AbstractApp implements CacheNavigationApp, WaypointNavigationApp, GeopointNavigationApp {
+
+    private static final String PARAM_NAME = "name";
+    private static final String PARAM_LAT = "lat";
+    private static final String PARAM_LON = "lon";
+    private static final String PREFIX = "osmand.api://";
+    private static final String GET_INFO = "get_info";
+    private static final String ADD_MAP_MARKER = "add_map_marker";
+
+    protected OsmAndApp() {
+        super(getString(R.string.cache_menu_osmand), null);
+    }
+
+    @Override
+    public boolean isInstalled() {
+        return ProcessUtils.isIntentAvailable(Intent.ACTION_VIEW, Uri.parse(PREFIX + GET_INFO));
+    }
+
+    @Override
+    public boolean isEnabled(@NonNull final Geocache cache) {
+        return cache.getCoords() != null;
+    }
+
+    @Override
+    public boolean isEnabled(@NonNull final Waypoint waypoint) {
+        return waypoint.getCoords() != null;
+    }
+
+    @Override
+    public void navigate(@NonNull final Activity activity, @NonNull final Geocache cache) {
+        if (cache.getCoords() != null) {
+            navigate(activity, cache.getCoords(), cache.getName());
+        } else {
+            ActivityMixin.showToast(activity, activity.getResources().getString(R.string.err_nav_no_coordinates));
+        }
+    }
+
+    @Override
+    public void navigate(@NonNull final Activity activity, @NonNull final Waypoint waypoint) {
+        if (waypoint.getCoords() != null) {
+            navigate(activity, waypoint.getCoords(), waypoint.getName());
+        } else {
+            ActivityMixin.showToast(activity, activity.getResources().getString(R.string.err_nav_no_coordinates));
+        }
+    }
+
+    @Override
+    public void navigate(@NonNull Activity activity, @NonNull Geopoint coords) {
+        navigate(activity, coords, activity.getResources().getString(R.string.osmand_marker_cgeo));
+    }
+
+    private void navigate(@NonNull final Activity activity, @NonNull final Geopoint coords, @NonNull final String markerName) {
+        final Map<String, String> params = new HashMap<>();
+        params.put(PARAM_LAT, String.valueOf(coords.getLatitude()));
+        params.put(PARAM_LON, String.valueOf(coords.getLongitude()));
+        params.put(PARAM_NAME, markerName);
+        activity.startActivity(buildIntent(params));
+    }
+
+    private Intent buildIntent(@Nullable final Map<String, String> parameters) {
+        final Uri uri = Uri.parse(getUriString(ADD_MAP_MARKER, parameters));
+        final Intent intent = new Intent(Intent.ACTION_VIEW, uri);
+        return intent;

```

Comment:
Do not make a variable just to return it right away. Just use `return new Intent(…)`.